nntrainer/graph/network_graph.cpp

```diff
@@ -287,23 +277,38 @@ int NetworkGraph::realizeActivationType(Layer &current) {
   ensureName(layer, current.getName());
 
   layer->setActivation(act);
-  layer->setNumInputs(1);
+  layer->setNumInputs(current.getNumInputs());
   layer->input_layers.clear();
   layer->input_layers.push_back(current.getName());
+  layer->setNumOutputs(current.getNumOutputs());
+  /** output layers for layer obj will be set in setOutputLayers() */
 
-  if (current.getNumOutputs() != 1)
-    return ML_ERROR_INVALID_PARAMETER;
+  updateConnectionName(current.getName(), layer->getName());
+  addLayerNode(layer);
 
-  layer->setNumOutputs(current.getNumOutputs());
-  layer->output_layers.clear();
-  for (unsigned int i = 0; i < current.getNumOutputs(); ++i)
-    layer->output_layers.push_back(current.output_layers[i]);
+  return status;
+}
 
-  current.setNumOutputs(1);
-  current.output_layers.clear();
-  current.output_layers.push_back(layer->getName());
+int NetworkGraph::realizeMultiOutputType(Layer &current) {
+  int status = ML_ERROR_NONE;
+  if (current.getNumOutputs() == 1)
+    return ML_ERROR_NONE;
+
+  std::shared_ptr<Layer> layer = nntrainer::createLayer(OutputLayer::type);
+  ensureName(layer, current.getName());
+
+  layer->setNumInputs(current.getNumInputs());
+  layer->input_layers.clear();
+  layer->input_layers.push_back(current.getName());
+  layer->setNumOutputs(current.output_layers.size());
```

Comment:
I wonder if setting output here is intended as the commit msg suggests differently.

Reply:
Yes, only the number of output layers is set. The output layers (connections in the graph) is set later.

Reply:
I see. Thank you for the explanation :)